tests/UniswapAnchoredViewTest.js

```diff
@@ -0,0 +1,172 @@
+  }
+  return {reporter, anchorMantissa, priceData, anchorPeriod, uniswapAnchoredView, tokenConfigs, postPrices, cToken};
+}
+
+describe('UniswapAnchoredView', () => {
+  let cToken, reporter, anchorMantissa, priceData, anchorPeriod, uniswapAnchoredView, tokenConfigs, postPrices;
+
+  describe('postPrices Unit Test', () => {
+    beforeEach(async done => {
+      ({reporter, anchorMantissa, priceData, uniswapAnchoredView, postPrices} = await setup({isMockedView: true}));
+      done();
+    })
+
+    it('should not update view if sender is not reporter', async () => {
+      const timestamp = time() - 5;
+      const nonSource = web3.eth.accounts.privateKeyToAccount('0x666ee777e72b8c042e05ef41d1db0f17f1fcb0e8150b37cfad6993e4373bdf10');
+      const tx = await postPrices(timestamp, [[['ETH', 91]]], ['ETH'], nonSource);
+      expect(tx.events.PriceGuarded).toBe(undefined);
+      expect(tx.events.PricePosted).toBe(undefined);
+      expect(await call(uniswapAnchoredView, 'prices', [keccak256('ETH')])).numEquals(0);
+    });
+
+    it('should update view if ETH price is within anchor bounds', async () => {
+      const timestamp = time() - 5;
+      await send(uniswapAnchoredView, 'setAnchorPrice', ['ETH', 91e6]);
+      const tx = await postPrices(timestamp, [[['ETH', 91]]], ['ETH']);
+
+      expect(tx.events.PriceGuarded).toBe(undefined);
+      expect(tx.events.PriceUpdated).not.toBe(undefined);
+      expect(await call(uniswapAnchoredView, 'prices', [keccak256('ETH')])).numEquals(91e6);
+      expect(await call(priceData, 'getPrice', [reporter.address, 'ETH'])).numEquals(91e6);
+    });
+
+    it('should not update view if ETH price is below anchor bounds', async () => {
+      // anchorMantissa is 1e17, so 10% tolerance
+      const timestamp = time() - 5;
+      await send(uniswapAnchoredView, 'setAnchorPrice', ['ETH', 89.9e6]);
+      const tx = await postPrices(timestamp, [[['ETH', 100]]], ['ETH']);
+
+      expect(tx.events.PriceGuarded).not.toBe(undefined);
+      expect(tx.events.PriceUpdated).toBe(undefined);
+      expect(await call(uniswapAnchoredView, 'prices', [keccak256('ETH')])).numEquals(0);
+      expect(await call(priceData, 'getPrice', [reporter.address, 'ETH'])).numEquals(100e6);
+    });
+
+    it('should not update view if ETH price is above anchor bounds', async () => {
+      // anchorMantissa is 1e17, so 10% tolerance
+      const timestamp = time() - 5;
+      await send(uniswapAnchoredView, 'setAnchorPrice', ['ETH', 110.1e6]);
+      const tx = await postPrices(timestamp, [[['ETH', 100]]], ['ETH']);
+
+      expect(tx.events.PriceGuarded).not.toBe(undefined);
+      expect(tx.events.PriceUpdated).toBe(undefined);
+      expect(await call(uniswapAnchoredView, 'prices', [keccak256('ETH')])).numEquals(0);
+      expect(await call(priceData, 'getPrice', [reporter.address, 'ETH'])).numEquals(100e6);
+    });
+
+    it.todo('test anchor with non-eth prices')
```

Comment:
This isn't valid I don't think

Reply:
hm?

Reply:
One half the Uniswap pair *must* be eth for the spec / code

Reply:
ha, yes just meant REP-ETH instead of ETH-USD